### PR TITLE
feat(interop) InteropL2StandardBridge

### PR DESCRIPTION
**Pull request by hamdiallam**
Analogous to #8254, an interop-enabled version of the standard bridge.

This implementation imposes a constraint that address space for l2-native tokens
remain consistent in the superchain to be minted/burned. (requires unification
of liquidity on L1)

This wraps up the remaining task for ethereum-optimism/interop#19

### CODE REVIEW

**Comment by hamdiallam** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8516?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `develop`
  * **PR #8462** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8462?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #8719** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8719?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #8516** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8516?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #8452** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8452?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8516?utm_source=stack-comment).

File: packages/contracts-bedrock/src/interop/InteropL2StandardBridge.sol [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+import { SafeCall } from "src/libraries/SafeCall.sol";
+import { ISemver } from "src/universal/ISemver.sol";
+
+import { ERC165Checker } from "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+
+/// @title InteropL2StandardBridge
+/// @notice InteropL2StandardBridge is a replacement of the L2StandardBridge that supports
+///         both L2-L2 and L2-L1 bridging.
+contract InteropL2StandardBridge is ISemver {
```
Comment (tynes) [inline packages/contracts-bedrock/src/interop/InteropL2StandardBridge.sol:19]: nit: naming is verbose

File: packages/contracts-bedrock/src/interop/InteropConstants.sol [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
```
Comment (tynes) [inline packages/contracts-bedrock/src/interop/InteropConstants.sol:3]: We have a file someplace that has chainids, i think its in the scripts directory, open to consolidate out of there into `libraries` and make it shared

File: packages/contracts-bedrock/src/interop/InteropL2StandardBridge.sol (hunk 1 above)
Comment (hamdiallam) [reply packages/contracts-bedrock/src/interop/InteropL2StandardBridge.sol:19]: yea not a fan of the name. Originally had it just `L2StandardBridge` & `NewL2StandardBridge` but the name collisions with forge and the bindings made that not work

**Review by protolambda — APPROVED** [review]

**Review by protolambda — APPROVED** [review]

**Pull request merged**